### PR TITLE
MF-539 - Improve Bootstrap Service documentation

### LINKED ISSUE #539: Improve Bootstrap Service documentation
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
Boostrapping (device onboarding) service was added recently. Official documentation is missing.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
Must-have

### PR DESCRIPTION
**Pull request by dborovcanin**
### What does this do?
This pull request updates Bootstrap service documentation.

### Which issue(s) does this PR fix/relate to?
This PR resolves #539.

### List any changes that modify/break current functionality
There are no such changes.

### Did you document any new/modified functionality?
Yes, this pull request is all about documenting.

### CODE REVIEW

File: docs/bootstrap.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+## Bootstrap
+
+`Bootstrapping` refers to a self-starting process that is supposed to proceed without external input.
+Mainflux platform supports bootstrapping process, but some of the preconditions need to be fulfilled in advance.
+
```
Comment (drasko) [inline docs/bootstrap.md:12]: Add architecture diagram and show where Bootstrap Service sits in the system.

Comment (drasko) [reply docs/bootstrap.md:12]: Something similar to this: https://medium.com/@vrmvrm/device-key-distribution-with-lightweight-m2m-36cdc12e5711

Comment (drasko) [reply docs/bootstrap.md:12]: Also - exmplain difference between Bootstrap and ordinary provisioning.

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — DISMISSED** [review]
LGTM

**Review by nmarcetic — DISMISSED** [review]

**Review by nmarcetic — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**